### PR TITLE
Hide IDataTransform Create methods

### PR DESCRIPTION
**Pull request by Ivanidzo4ka**
Step towards https://github.com/dotnet/machinelearning/issues/1995.
Also add proper comments for Image transforms.
Remove Create methods in Onnx and Tensorflow (replace them with proper constructors).

### CODE REVIEW

File: src/Microsoft.ML.OnnxTransform.StaticPipe/DnnImageFeaturizerStaticExtensions.cs [hunk 1]
```diff
@@ -8,9 +8,8 @@
 using Microsoft.ML.Data;
 using Microsoft.ML.StaticPipe;
 using Microsoft.ML.StaticPipe.Runtime;
-using Microsoft.ML.Transforms;
 
-namespace Microsoft.ML.OnnxTransform.StaticPipe
+namespace Microsoft.ML.Transforms.StaticPipe
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.OnnxTransform.StaticPipe/DnnImageFeaturizerStaticExtensions.cs]: >Transforms [](start = 23, length = 10)

Reason why I make this change is because I don't like what I have to do this:
`new Transforms.OnnxTransform(...);`
instead of just 
` new OnnxTransform(...);`
if we have Microsoft.ML in usings. (see OnnxTransformTests.cs) 

Shall we just call it Microsoft.ML.StaticPipe without any additional sub namespaces?

File: src/Microsoft.ML.Data/Transforms/KeyToValue.cs [hunk 2]
```diff
@@ -109,9 +108,7 @@ public static IDataTransform Create(IHostEnvironment env, Arguments args, IDataV
             return transformer.MakeDataTransform(input);
         }
 
-        /// <summary>
-        /// Factory method for SignatureLoadModel.
-        /// </summary>
+        // Factory method for SignatureLoadModel.
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/Transforms/KeyToValue.cs]: >// Factory method for SignatureLoadModel. [](start = 8, length = 41)

I see you doing this multiple places, is there any reason for it?

If we instead *kept* it an XML comment, we could improve this by having a `<see` tag on it, which overall seems like a good idea. We're not doing this now, but it seems like an unforced regression into a worse, not a better, place.

File: src/Microsoft.ML.Data/Transforms/NormalizeColumn.cs [hunk 3]
```diff
@@ -283,10 +281,8 @@ public static IDataTransform Create(IHostEnvironment env, MinMaxArguments args,
             return normalizer.Fit(input).MakeDataTransform(input);
         }
 
-        /// <summary>
-        /// Public create method corresponding to SignatureDataTransform.
-        /// </summary>
-        public static IDataTransform Create(IHostEnvironment env, MeanVarArguments args, IDataView input)
+        // Factory method corresponding to SignatureDataTransform.
+        internal static IDataTransform Create(IHostEnvironment env, MeanVarArguments args, IDataView input)
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/Transforms/NormalizeColumn.cs]: >internal [](start = 8, length = 8)

If the comment is correct, then it should suffice for it to be `private`, right? So is the comment incorrect?

File: src/Microsoft.ML.Data/Transforms/OneToOneTransformerBase.cs [hunk 4]
```diff
@@ -15,7 +15,7 @@ public abstract class OneToOneTransformerBase : RowToRowTransformerBase
     {
         protected readonly (string input, string output)[] ColumnPairs;
 
-        protected OneToOneTransformerBase(IHost host, (string input, string output)[] columns) : base(host)
+        protected OneToOneTransformerBase(IHost host, params (string input, string output)[] columns) : base(host)
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/Transforms/OneToOneTransformerBase.cs]: >params [](start = 54, length = 6)

Is this related? It might be a good idea, just not sure?

File: src/Microsoft.ML.Data/Transforms/NormalizeColumn.cs (hunk 3 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Data/Transforms/NormalizeColumn.cs]: Everytime we have Entrypoint for transform I have to mark this method as internal.
As alternative I can refactor conversion of argument class to ColumnInfo array and use it here and in entrypoint, but oh, boy, it would be lot of changes across codebase.

---
In reply to: [245457063](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245457063) [](ancestors = 245457063)

File: src/Microsoft.ML.Data/Transforms/KeyToValue.cs (hunk 2 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Data/Transforms/KeyToValue.cs]: only reason why we had <summary> tag because it was requirement for public method, and since it's no longer public I just prefer mark this methods as comments since 
a) This is most common pattern in our codebase.
b) I think about this as more as internal message for developer: "Hey, don't delete this one, even if no one uses it, we actually call it via our DI system." Summary is more like a description of what is going on in this method, and this is more like a mark/internal comment.

---
In reply to: [245456969](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245456969) [](ancestors = 245456969)

File: src/Microsoft.ML.OnnxTransform/OnnxTransform.cs [hunk 5]
```diff
@@ -100,32 +100,8 @@ private static VersionInfo GetVersionInfo()
             loaderAssemblyName: typeof(OnnxTransform).Assembly.FullName);
         }
 
-        public static IDataTransform Create(IHostEnvironment env, IDataView input, string modelFile, int? gpuDeviceId = null, bool fallbackToCpu = false)
```
Comment (TomFinley) [inline src/Microsoft.ML.OnnxTransform/OnnxTransform.cs]: >public static IDataTransform Create(IHostEnvironment env, IDataView input, string modelFile, int? gpuDeviceId = null, bool fallbackToCpu = false) [](start = 8, length = 145)

This is an "odd" method, so it seems good to remove, so are we instead working via the extension methods and `MLContext`?

File: src/Microsoft.ML.Data/Transforms/OneToOneTransformerBase.cs (hunk 4 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Data/Transforms/OneToOneTransformerBase.cs]: I think I replaced one of Create methods with constructor which accept only one input and output column, and I didn't like idea what I need to cast that pair into array of pairs. 
This is why I changed it. Not sure that constructor is still in part of codereview, but I don't think it's in general bad thing to have.

---
In reply to: [245457186](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245457186) [](ancestors = 245457186)

File: src/Microsoft.ML.OnnxTransform.StaticPipe/DnnImageFeaturizerStaticExtensions.cs (hunk 1 above)
Comment (TomFinley) [reply src/Microsoft.ML.OnnxTransform.StaticPipe/DnnImageFeaturizerStaticExtensions.cs]: I'd expect that static pipe would be something we'd nail down later, but I'd *expect* that inputs to to staticpipe would just be accomplished by taking a `PiplineColumn` instance, and applying the transform to it. There should not be any `new Transforms.OnnxTransform` in the static pipeline at all, since the static pipeline is more declarative. So I don't like either of the things you've described, if indeed that is now it iss set up. If that's not how it was set up, well, that's bad. :) So that would be something to file and issue on and fix. (Though I think static pipeline is something we're going to refine more post v1.0.)

---
In reply to: [245456891](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245456891) [](ancestors = 245456891)

File: src/Microsoft.ML.OnnxTransform/OnnxTransform.cs (hunk 5 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.OnnxTransform/OnnxTransform.cs]: I'm not sure about extension methods, but in tests I switch to using `OnnxTransform()` constructor with same parameters and then calling `Transform` method on DataView.
I think it's better to get rid of this Create methods, and use either estimator, or since estimator is trivial, just call directly transform consturctor.

---
In reply to: [245457717](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245457717) [](ancestors = 245457717)

File: src/Microsoft.ML.OnnxTransform.StaticPipe/DnnImageFeaturizerStaticExtensions.cs (hunk 1 above)
Comment (TomFinley) [reply src/Microsoft.ML.OnnxTransform.StaticPipe/DnnImageFeaturizerStaticExtensions.cs]: Actually, from what I see below, it *is* set up as an extension method on the columns. So what is the problem?

---
In reply to: [245458003](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245458003) [](ancestors = 245458003,245456891)

File: src/Microsoft.ML.TensorFlow/TensorflowTransform.cs [hunk 6]
```diff
@@ -178,33 +178,61 @@ private static VersionInfo GetVersionInfo()
...
-        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
-        /// <param name="model">Path to the TensorFlow model. </param>
-        /// <param name="names">Name of the output column(s). Keep it same as in the Tensorflow model.</param>
-        /// <param name="source">Name of the input column(s). Keep it same as in the Tensorflow model.</param>
-        public static IDataTransform Create(IHostEnvironment env, IDataView input, string model, string[] names, string[] source)
+        /// <param name="env">The environment to use.</param>
+        /// <param name="modelFile">Model file path.</param>
+        /// <param name="inputColumn">The name of the input data column. Must match model input name.</param>
+        /// <param name="outputColumn">The output columns to generate. Names must match model specifications. Data types are inferred from model.</param>
+        public TensorFlowTransform(IHostEnvironment env, string modelFile, string inputColumn, string outputColumn)
```
Comment (TomFinley) [inline src/Microsoft.ML.TensorFlow/TensorflowTransform.cs]: >TensorFlowTransform [](start = 15, length = 19)

The naming I find incomprehensible, similar to the ONNX thing. It is called a `Transform`, yet it appears to be a `Transformer`? Do we know what is going on here?

File: src/Microsoft.ML.OnnxTransform.StaticPipe/DnnImageFeaturizerStaticExtensions.cs (hunk 1 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.OnnxTransform.StaticPipe/DnnImageFeaturizerStaticExtensions.cs]: If you have your code and you use `using Microsoft.ML;` in it (which we do all the time), I can't longer just create OnnxTransform object via `new OnnxTransform(...)`, because you will get collision of namespace and object.
So you have to directly specify what your transform is in Transform namespace like this: `Transforms.OnnxTransform` which I found annoying.

---
In reply to: [245458074](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245458074) [](ancestors = 245458074,245458003,245456891)

File: src/Microsoft.ML.Data/Transforms/KeyToValue.cs (hunk 2 above)
Comment (TomFinley) [reply src/Microsoft.ML.Data/Transforms/KeyToValue.cs]: <summary> tags are a requirement for public methods. That ***does not make them a bad thing for private or internal methods***, especially if those comments explicitly reference other structures in the codebase, as this does.

Why would a *developer* prefer this comment, to the old comment? If you had instead had a `<see` tag, as suggested above, that developer could even F12 on it to see what the heck we're talking about. Right?

---
In reply to: [245457620](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245457620) [](ancestors = 245457620,245456969)

File: src/Microsoft.ML.OnnxTransform.StaticPipe/DnnImageFeaturizerStaticExtensions.cs (hunk 1 above)
Comment (TomFinley) [reply src/Microsoft.ML.OnnxTransform.StaticPipe/DnnImageFeaturizerStaticExtensions.cs]: Oh I see. Yes I agree then, I didn't know you were talking about namespace/object collision.

Ultimately all static extensions should be in `Microsoft.ML` namespace (since we must accept that inclusion of a specific nuget is license to use it, similar to our attitude about `MLContext` extensions), but not right now.

---
In reply to: [245458427](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245458427) [](ancestors = 245458427,245458074,245458003,245456891)

File: src/Microsoft.ML.Data/Transforms/NormalizeColumn.cs (hunk 3 above)
Comment (TomFinley) [reply src/Microsoft.ML.Data/Transforms/NormalizeColumn.cs]: Oh entrypoints. Fair enough.

---
In reply to: [245457335](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245457335) [](ancestors = 245457335,245457063)

File: src/Microsoft.ML.OnnxTransform/OnnxTransform.cs (hunk 5 above)
Comment (TomFinley) [reply src/Microsoft.ML.OnnxTransform/OnnxTransform.cs]: Yes, probably.

---
In reply to: [245458012](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245458012) [](ancestors = 245458012,245457717)

**Comment by TomFinley** [comment]
>     public sealed class OnnxTransform : RowToRowTransformerBase

This naming is very strange... it is called OnnxTransform, but, is it a `IDataTransform`, or an `ITransformer`?

---
Refers to: src/Microsoft.ML.OnnxTransform/OnnxTransform.cs:58 in 185099d. [](commit_id = 185099d47867a4f18c4c0f6e56857636daf7e01f, deletion_comment = False)

File: src/Microsoft.ML.Data/Transforms/OneToOneTransformerBase.cs (hunk 4 above)
Comment (TomFinley) [reply src/Microsoft.ML.Data/Transforms/OneToOneTransformerBase.cs]: All right, thanks Ivan.

---
In reply to: [245457798](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245457798) [](ancestors = 245457798,245457186)

File: src/Microsoft.ML.Data/Transforms/KeyToValue.cs (hunk 2 above)
Comment (TomFinley) [reply src/Microsoft.ML.Data/Transforms/KeyToValue.cs]: This should probably just be a simple search-replace, I think, to put it into proper form. At least, so I'd think.

---
In reply to: [245458497](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245458497) [](ancestors = 245458497,245457620,245456969)

**Comment by Ivanidzo4ka** [comment]
>     public sealed class OnnxTransform : RowToRowTransformerBase

It's a `ITransformer`, I actually converted it to `ITransformer` some time ago, but didn't rename it.
I push another iteration with renaming for TensorFlow, Onnx, and Image transforms.

---
In reply to: [451613124](https://github.com/dotnet/machinelearning/pull/2031#issuecomment-451613124) [](ancestors = 451613124)

---
Refers to: src/Microsoft.ML.OnnxTransform/OnnxTransform.cs:58 in 185099d. [](commit_id = 185099d47867a4f18c4c0f6e56857636daf7e01f, deletion_comment = False)

File: src/Microsoft.ML.TensorFlow/TensorflowTransform.cs (hunk 6 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.TensorFlow/TensorflowTransform.cs]: It's a transformer now

---
In reply to: [245458201](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245458201) [](ancestors = 245458201)

File: src/Microsoft.ML.Data/Transforms/KeyToValue.cs (hunk 2 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Data/Transforms/KeyToValue.cs]: I know I can't win argument with you so I reverted my changes to summary comments. Although I'm still adding comments for other places since this is common pattern in our code.
I hope this is not a blocker, since main intent of this PR is to hide IDataTransforms methods.

---
In reply to: [245459044](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245459044) [](ancestors = 245459044,245458497,245457620,245456969)

File: src/Microsoft.ML.Data/Transforms/ColumnCopying.cs [hunk 7]
```diff
@@ -39,7 +39,7 @@ public sealed class ColumnCopyingEstimator : TrivialEstimator<ColumnCopyingTrans
         {
         }
 
-        public ColumnCopyingEstimator(IHostEnvironment env, params (string source, string name)[] columns)
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/Transforms/ColumnCopying.cs]: >ColumnCopyingEstimator [](start = 15, length = 22)

I'll bug the next person about XML comments, or do it:)

File: src/Microsoft.ML.Data/Transforms/ColumnCopying.cs [hunk 8]
```diff
@@ -110,8 +110,8 @@ public sealed class Arguments : TransformInputBase
             public Column[] Column;
         }
 
-        // Factory method to create from arguments
-        public static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+        // Factory method corresponding to SignatureDataTransform.
+        internal static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/Transforms/ColumnCopying.cs]: >internal [](start = 8, length = 8)

double-checking: can't be private?

File: src/Microsoft.ML.Data/Transforms/ColumnCopying.cs (hunk 7 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Data/Transforms/ColumnCopying.cs]: Would you trust me if I say what I plan to do that in other PR?
This one is purely IDataTransform hiding.

---
In reply to: [246154879](https://github.com/dotnet/machinelearning/pull/2031#discussion_r246154879) [](ancestors = 246154879)

Comment (sfilipi) [reply src/Microsoft.ML.Data/Transforms/ColumnCopying.cs]: yep!

---
In reply to: [246156509](https://github.com/dotnet/machinelearning/pull/2031#discussion_r246156509) [](ancestors = 246156509,246154879)

File: src/Microsoft.ML.ImageAnalytics/ImageResizerTransform.cs [hunk 9]
```diff
@@ -423,20 +452,39 @@ protected override Delegate MakeGetter(Row input, int iinfo, Func<int, bool> act
...
-        public ImageResizingEstimator(IHostEnvironment env, string inputColumn, string outputColumn,
-            int imageWidth, int imageHeight, ImageResizerTransform.ResizingKind resizing = ImageResizerTransform.ResizingKind.IsoCrop, ImageResizerTransform.Anchor cropAnchor = ImageResizerTransform.Anchor.Center)
-            : this(env, new ImageResizerTransform(env, inputColumn, outputColumn, imageWidth, imageHeight, resizing, cropAnchor))
+        /// <summary>
+        /// Estimator which resize images.
+        /// </summary>
+        /// <param name="env">The host environment.</param>
+        /// <param name="input">Name of the input column.</param>
+        /// <param name="output">Name of the column resulting from the transformation of <paramref name="input"/>.</param>
+        /// <param name="imageWidth">Widht of resized image.</param>
```
Comment (sfilipi) [inline src/Microsoft.ML.ImageAnalytics/ImageResizerTransform.cs]: >Widht [](start = 37, length = 5)

width.

File: src/Microsoft.ML.ImageAnalytics/ImageResizerTransform.cs [hunk 10]
```diff
@@ -423,20 +452,39 @@ protected override Delegate MakeGetter(Row input, int iinfo, Func<int, bool> act
...
+    /// <summary>
+    /// Estimator which resize images.
+    /// </summary>
+    public sealed class ImageResizingEstimator : TrivialEstimator<ImageResizerTransformer>
     {
-        public ImageResizingEstimator(IHostEnvironment env, string inputColumn, string outputColumn,
-            int imageWidth, int imageHeight, ImageResizerTransform.ResizingKind resizing = ImageResizerTransform.ResizingKind.IsoCrop, ImageResizerTransform.Anchor cropAnchor = ImageResizerTransform.Anchor.Center)
-            : this(env, new ImageResizerTransform(env, inputColumn, outputColumn, imageWidth, imageHeight, resizing, cropAnchor))
+        /// <summary>
+        /// Estimator which resize images.
```
Comment (sfilipi) [inline src/Microsoft.ML.ImageAnalytics/ImageResizerTransform.cs]: >resize [](start = 28, length = 6)

resizes

File: src/Microsoft.ML.ImageAnalytics/ImageResizerTransform.cs [hunk 11]
```diff
@@ -423,20 +452,39 @@ protected override Delegate MakeGetter(Row input, int iinfo, Func<int, bool> act
...
+        /// </summary>
+        /// <param name="env">The host environment.</param>
+        /// <param name="input">Name of the input column.</param>
+        /// <param name="output">Name of the column resulting from the transformation of <paramref name="input"/>.</param>
+        /// <param name="imageWidth">Widht of resized image.</param>
+        /// <param name="imageHeight">Height of resized image.</param>
+        /// <param name="resizing">What resize method to use.</param>
```
Comment (sfilipi) [inline src/Microsoft.ML.ImageAnalytics/ImageResizerTransform.cs]: >resize method [](start = 40, length = 13)

<see cref="ImageResizerTransformer.ResizingKind"/>

File: src/Microsoft.ML.OnnxTransform.StaticPipe/DnnImageFeaturizerStaticExtensions.cs (hunk 1 above)
Comment (sfilipi) [reply src/Microsoft.ML.OnnxTransform.StaticPipe/DnnImageFeaturizerStaticExtensions.cs]: But that happens only when ppl use both static and dynamic, right? Our tests do so, but idk how many ppl will mix ant match static and dynamic. 

---
In reply to: [245458713](https://github.com/dotnet/machinelearning/pull/2031#discussion_r245458713) [](ancestors = 245458713,245458427,245458074,245458003,245456891)

File: src/Microsoft.ML.Transforms/MissingValueHandlingTransformer.cs [hunk 12]
```diff
@@ -115,7 +116,7 @@ public bool TryUnparse(StringBuilder sb)
         /// <param name="name">Name of the output column.</param>
         /// <param name="source">Name of the column to be transformed. If this is null '<paramref name="name"/>' will be used.</param>
         /// <param name="replaceWith">The replacement method to utilize.</param>
-        public static IDataTransform Create(IHostEnvironment env, IDataView input, string name, string source = null, ReplacementKind replaceWith = ReplacementKind.DefaultValue)
+        public static IDataView Create(IHostEnvironment env, IDataView input, string name, string source = null, ReplacementKind replaceWith = ReplacementKind.DefaultValue)
```
Comment (sfilipi) [inline src/Microsoft.ML.Transforms/MissingValueHandlingTransformer.cs]: >public [](start = 8, length = 6)

Internal?

File: src/Microsoft.ML.Transforms/Text/TextFeaturizingEstimator.cs [hunk 13]
```diff
@@ -279,8 +281,10 @@ public TransformApplierParams(TextFeaturizingEstimator  parent)
             advancedSettings?.Invoke(AdvancedSettings);
 
             _dictionary = null;
-            _wordFeatureExtractor = new NgramExtractorTransform.NgramExtractorArguments();
-            _charFeatureExtractor = new NgramExtractorTransform.NgramExtractorArguments() { NgramLength = 3, AllLengths = false };
+            if (AdvancedSettings.UseWordExtractor)
```
Comment (sfilipi) [inline src/Microsoft.ML.Transforms/Text/TextFeaturizingEstimator.cs]: > if (AdvancedSettings.UseWordExtractor) [](start = 11, length = 39)

Pr title should have been: Hide the Create methods, change the name of the ImageTransformers to match the conventions used elsewhere, and fix every bug, whitespace and whatnot i see as i work :D

File: src/Microsoft.ML.Transforms/Text/WordHashBagProducingTransform.cs [hunk 14]
```diff
@@ -408,21 +400,8 @@ public sealed class Arguments : ArgumentsBase
...
-
-            // Args for the Hash function with multiple columns
-            var hashArgs =
-                new HashingTransformer.Arguments
-                {
-                    HashBits = 31,
-                    Seed = args.Seed,
-                    Ordered = false,
-                    Column = hashColumns.ToArray(),
-                    InvertHash = args.InvertHash
```
Comment (sfilipi) [inline src/Microsoft.ML.Transforms/Text/WordHashBagProducingTransform.cs]: >args.InvertHash [](start = 33, length = 15)

are we losing this and the args.Seed?

File: test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs [hunk 15]
```diff
@@ -755,23 +747,8 @@ public void EntryPointPipelineEnsemble()
...
-                    {
-                        new ValueToKeyMappingTransformer.Column()
-                        {
-                            Name = "Label",
-                            Source = "Label",
-                            Sort = ValueToKeyMappingTransformer.SortOrder.Value
-                        }
-                    }
-                }, data);
+                data = new ColumnConcatenatingTransformer(Env, "Features", new[] { "Features1", "Features2" }).Transform(data);
```
Comment (sfilipi) [inline test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs]: >data = new ColumnConcatenatingTransformer(Env, "Features", new[] { "Features1", "Features2" }).Transform(data); [](start = 16, length = 111)

me looking at this and thinking that we should make the Transformer constructors internal.

Is there any benefit to leaving them public? (i know we've had the conversation for the Predictors/ModelParams and that made sense.)

If it is more than a yes/no answer, i can create an issue and we can discuss it there.

Comment (Ivanidzo4ka) [reply test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs]: I think idea was if it's a trivial estimator, it's ok to have public constructor for it's transformer.
Estimator is become pointless for them unless you try to build chain of estimators.

---
In reply to: [246171719](https://github.com/dotnet/machinelearning/pull/2031#discussion_r246171719) [](ancestors = 246171719)

Comment (sfilipi) [reply test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs]: ok, thanks.

---
In reply to: [246172551](https://github.com/dotnet/machinelearning/pull/2031#discussion_r246172551) [](ancestors = 246172551,246171719)
